tendermint/src/block/height.rs

```diff
@@ -7,25 +7,27 @@ use std::{
 
 /// Block height for a particular chain (i.e. number of blocks created since
 /// the chain began)
+/// 
+/// A height of 0 represents a chain which has not yet produced a block.
 #[derive(Copy, Clone, Eq, Hash, PartialEq, PartialOrd, Ord)]
 pub struct Height(u64);
 
 impl Height {
     /// Convert `u64` to block height.
-    ///
-    /// Note that 0 is not a valid block height.
+    /// 
+    /// Note that this method will never error and is just for backwards
+    /// compatibility.
     pub fn try_from_u64(n: u64) -> Result<Self, Error> {
-        // Minimum height is 1
-        if n > 0 {
-            Ok(Height(n))
-        } else {
-            Err(ErrorKind::OutOfRange.into())
-        }
+        Ok(Height(n))
     }
 
     /// Convert `i64` (used in e.g. Amino messages) to block height.
     pub fn try_from_i64(n: i64) -> Result<Self, Error> {
```

Comment:
Perhaps this should be a proper `TryFrom` now

Reply:
Should I also get rid of the existing panic-able `From<i64>`?

Reply:
That'd probably be for the best (at the time this was written `TryFrom`/`TryInto` weren't stable)

Reply:
✅